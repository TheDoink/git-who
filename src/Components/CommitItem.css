.commitSha {
	font-size: .75em;
}