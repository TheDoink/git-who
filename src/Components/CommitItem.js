import React, { Component } from 'react';
import NameLink from './NameLink';
import './RepoItem.css'; // There's some reuse of styles here
import './CommitItem.css';

class RepoItem extends Component {

  // Quick function to error check and see who owns the item
  getAuthor() {
    let ret = "";
    if(this.props.commit && this.props.commit.author) {
      ret = this.props.commit.author.login;
    }
    
    return ret;
  }

  getMessage() {
    let ret = "";
    if(this.props.commit && this.props.commit.commit) {
      ret = this.props.commit.commit.message
    }
    
    return ret;
  }
  
  render() {
    
    return (
        <div className="repoItem">
          <div className="repoSuperScript">
            <div className="commitAuthor">
              <NameLink userName={this.getAuthor.bind(this)()}></NameLink>
            </div>
          </div>
          <div className="repoDescription">{this.getMessage.bind(this)()}</div>
          <span className="repoSubScript">
            <span className="commitSha">{this.props.commit.sha}</span>
          </span>
        </div>
    );
  }
}

export default RepoItem;
