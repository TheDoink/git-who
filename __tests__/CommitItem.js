// var NameLink = require('../src/Components/NameLink');
import React from 'react';
import renderer from 'react-test-renderer';
import CommitItem from '../src/Components/CommitItem';

const exampleCommit = {
    "sha": "a5eca57f4a29e99d17e296ee6afe5ff4c64ff931",
    "commit": {
      "author": {
        "name": "Andrey Lushnikov"
      },

      "message": "Add proxy server example (#427)"
    },
    "author": {
      "login": "aslushnikov",
      "id": 746130
    },
    "committer": {
      "login": "ebidel",
      "id": 238208,
    },
    "parents": [
      {
        "sha": "281db8feccfda0f887fafd31fc4f9dcff414d531",
        "url": "https://api.github.com/repos/GoogleChrome/puppeteer/commits/281db8feccfda0f887fafd31fc4f9dcff414d531",
        "html_url": "https://github.com/GoogleChrome/puppeteer/commit/281db8feccfda0f887fafd31fc4f9dcff414d531"
      }
    ]
  };

it('lines up with the snapshot', () => {
  let testMe = renderer.create(<CommitItem key="35387538" commit={exampleCommit}></CommitItem>).toJSON();

  expect(testMe).toMatchSnapshot();
  // testMe.getGist('d2eace7977e2aeaf942de0f8b3e7275c');

  
});
