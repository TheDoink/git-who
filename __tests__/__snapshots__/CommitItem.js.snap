// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lines up with the snapshot 1`] = `
<div
  className="repoItem"
>
  <div
    className="repoSuperScript"
  >
    <div
      className="commitAuthor"
    >
      <span>
        <a
          href="#"
          onClick={[Function]}
        >
          aslushnikov
        </a>
      </span>
    </div>
  </div>
  <div
    className="repoDescription"
  >
    Add proxy server example (#427)
  </div>
  <span
    className="repoSubScript"
  >
    <span
      className="commitSha"
    >
      a5eca57f4a29e99d17e296ee6afe5ff4c64ff931
    </span>
  </span>
</div>
`;
